src/visualizers/widgets/MonacoEditor/MonacoEditorWidget.js

```diff
@@ -0,0 +1,476 @@
+    'jquery-contextMenu',
+    'css!./styles/MonacoEditorWidget.css',
+], function (
+    ComponentSettings,
+    MonacoLanguages,
+    ThemeList,
+    Chance
+) {
+    'use strict';
+
+    MonacoLanguages = JSON.parse(MonacoLanguages);
+    ThemeList = JSON.parse(ThemeList);
+
+    const WIDGET_CLASS = 'monaco-editor';
+    const DISPLAY_MENU_CLASS = 'display-menu';
+    const DEFAULT_THEMES = ['vs', 'vs-dark', 'hc-black'];
+    const DEFAULT_COLORS = {
+        'vs': '#fffffe',
+        'vs-dark': '#1e1e1e',
+        'hc-black': '#000000'
+    };
+
+    const AVAILABLE_KEYBINDINGS = ['default', 'vim'];
+
+    const TAB_NAV = `<ul class="nav nav-tabs nav-display-menu">
+                        <li class="pull-right nav-item">
+                        </li>
+                        <li data-role="button" class="pull-right nav-item">
+                            <button class="btn btn-link ${DISPLAY_MENU_CLASS}">
+                               <i class="glyphicon glyphicon-cog"></i>
+                               Display Settings
+                            </button>
+                        </li>
+                     </ul>`;
+
+    const chance = new Chance();
+
+    function MonacoEditorWidget(logger, container, config = {}) {
+        this._logger = logger.fork('Widget');
+        const modelURI = config.modelURI ? config.modelURI.split(/\./).shift() : 'inmemory://code';
+        const extension = config.modelURI ? config.modelURI.split(/\./).pop() : '.py';
+        this.modelURI = modelURI + chance.string({length: 5}) + extension;
+        this.monacoURI = monaco.Uri.parse(this.modelURI);
+        this.language = config.language || 'python';
+        this.keyBindingAdapter = null;
+
+        this._el = container;
+        this._el.css({height: '100%'});
+        this._el.append($(TAB_NAV));
+
+        this.$editor = $('<div/>');
+        this.$status = $('<div/>');
+        this.$editor.css({height: '100%'});
+        this._el.append(this.$editor[0]);
+        this._el.append(this.$status[0]);
+
+        this.readOnly = config.readOnly || false;
+
+        // register context menu to display settings and ComponentSettings
+        // Currently there's no submenu in monaco-editor and context menu is used
+        // to add display settings. See https://github.com/microsoft/monaco-editor/issues/1947
+        this._registerContextMenu();
+
+        // register languages to the monaco editor
+        this._registerLanguages();
+
+        // Create editor with value provided by constructor
+        const value = config.value || "def dummy_python_func():\n\tpass";
+        this.model = monaco.editor.getModel(this.monacoURI) ||
+            monaco.editor.createModel(
+                value,
+                this.language,
+                this.monacoURI
+            );
+        this.model.updateOptions({
+            tabSize: 4,
+            insertSpaces: true
+        });
+
+        const dontDisplayMiniMap = config.dontDisplayMiniMap || false;
```

Comment:
As a minor comment, it might be easier to understand this if it is `displayMinimap` rather than the negation of it.